src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -602,16 +603,22 @@ private boolean reconciliationComplete() {
     @Override
     public void run() {
       log.info("Current persistent state:");
-      log.info(String.format("JournalNodes: %s", persistentState.getJournalNodes()));
-      log.info(String.format("NameNodes: %s", persistentState.getNameNodes()));
+      log.info(String.format("JournalNodes: %s %s", persistentState.getJournalNodes(),
+          persistentState.getJournalNodeTaskNames()));
+      log.info(String.format("NameNodes: %s %s", persistentState.getNameNodes(),
+          persistentState.getNameNodeTaskNames()));
       log.info(String.format("DataNodes: %s", persistentState.getDataNodes()));
 
-      Collection<String> taskIds = persistentState.getAllTaskIds();
-      HashMap<String, Protos.TaskStatus> runningTaskIds = liveState.getRunningTasks();
+      Set<String> taskIds = persistentState.getAllTaskIds();
+      Set<String> runningTaskIds = liveState.getRunningTasks().keySet();
+
       for (String taskId : taskIds) {
-        if (!runningTaskIds.containsKey(taskId))
+        if (taskId != null && !runningTaskIds.contains(taskId)) {
```

Comment:
Would we ever have a null taskId in persistentState.getAllTaskIds? Seems like more of an assert than an if condition.